src/smart-components/Channel/context/dux/reducers.js

```diff
@@ -168,23 +168,25 @@ export default function reducer(state, action) {
         ],
       };
     case actionTypes.SEND_MESSAGEGE_SUCESS: {
-      const newMessages = state.allMessages.map((m) => (
-        compareIds(m.reqId, action.payload.reqId) ? action.payload : m
+      const message = action.payload;
+      const filteredMessages = state.allMessages.filter((m) => (
+        m?.reqId !== message?.reqId
       ));
-      [...newMessages].sort((a, b) => (
-        (
-          a.sendingStatus
-          && b.sendingStatus
-          && a.sendingStatus === SUCCEEDED
-          && (
-            b.sendingStatus === PENDING
-            || b.sendingStatus === FAILED
-          )
-        ) ? -1 : 1
+      const pendingIndex = filteredMessages.findIndex((msg) => (
+        msg?.sendingStatus === 'pending' || msg?.sendingStatus === 'failed'
       ));
       return {
         ...state,
-        allMessages: newMessages,
+        allMessages: pendingIndex > -1
+          ? [
+            ...filteredMessages.slice(0, pendingIndex),
+            message,
```

Comment:
Ddi you add this for some QA ticket?
Can we add a test for this? mentioning the situation that we are trying to avoid?